### DISCUSSION TITLE
chronological data management

### DISCUSSION THREAD

**Discussion by urapadmin** [arch-kiosk/arch-kiosk-office#1057] [users: ideas for new features] [answered]
Currently Kiosk only stores the data in it's most recent state. 
It is not possible to go back in time and look at changes from the past. While this is already a planned feature for the future of the file repository, it has never been discussed for the recording data itself. 

So: Do we see a reason and use in storing all data chronologically and compare former changes with the last recent version?

**Comment by luizaogs** [comment]
I'm about to say a bunch of nonsense, I suspect, but I was looking through the other discussion topics and it strikes me that this is a difference from paper recording? Theoretically if one is recording on paper one can flip through a notebook to earlier interpretations, or even if you have an old context form that has become outdated for some reason (or that was incorrectly recorded), one shouldn't throw that away. So then you have a record of sorts. But if the general approach to recording here is not deleting records but rather adding new ones, I suppose that record already exists? So maybe I'm misunderstanding the point.

**Comment by urapadmin** [comment]
I did not know that one has old locus sheets with former interpretations. It does not sound a systematical approach even on paper, though, does it? You might have accidentally an older version of you data, but it is not a natural effect of the approach that you would have. If you just add something to your locus sheet you would not know that you did that on a different day, unless you mention that, would you?

When it comes to interpretation (which we have totally neglected in the current recording) I would suggest that we at least should have a list of interpretation remarks as we have in the daily narratives to NOT override older ideas or produce a cumbersome text blob. But that is an individual design decision. This here was meant to discuss whether we want to be able to go back in time in general. Currently you cannot see what the state of anything was exactly on a certain day in the past. 

Somebody might have changed the color or size or something of a collected material later and you can't go back to before that change.

**Comment by luizaogs** [comment]
Interpretation was a random example on my part, I just meant that paper records usually do not get thrown away in excavations so then you technically have a means of going back in time (at least I have never thrown out forms when supervising units because that seems like a terrible idea). But no, I guess you wouldn't know when you had added a specific thing to a locus form unless you dated that specific change. And I wouldn't say that outdated forms are necessarily ever consulted either, which I guess might speak to the general usefulness of this, unless there is a mistake that needs to be checked. Then, yes, useful.

Would combining this with the director's view be of interest/useful in any way, or would you have to build something completely new?

**Comment by lbestock** [comment]
I am not convinced the utility of going back matches any effort it would take to introduce it. Interpretation is an interesting one because it is really the only place where it really makes sense to see previous stuff. But the relationship to paper recording is instructive - interpretation doesn't show up on locus sheets. Locus sheets are being misused if they are interpretive; it's a separate tab for us for a reason. Narratives and journaling often catch a lot of the growth of ideas. I would be open to including the same structure we have for narratives in the interpretation tab, but not until we have a bigger discussion about interpretation and analysis. This is really about post-field things, not recording. I did find myself abusing the interpretation tab when I was working on publishing Uronarti stuff and figuring out loci but that was really wrong - it meant that interpretations that were of groups of loci and their relationships ended up in the record of a single locus. So, needs more thought and a place separate from the field recording.

Aside from interpretation, I really can't see it being useful, honestly. If a value is changed, it is changed because the registrar used a Munsell book to get the color while the excavator just eyeballed it - a refining that is a correction, not an interpretation where the first one should be saved. Will there be occasional accidents? Yes, but how could you know the original record was more accurate than the revision? Just having both of them doesn't help you decide which is better unless you have the thing itself in front of you.

**Comment by urapadmin** [comment, marked as answer]
okay. In that case using a temporal database structure that keeps every point in time automatically is not required. Single record type where it would still be useful to rather add things than modify them (as with narratives) will be single design decisions but the underlying database handling does not need the additional complexity. Which is a relief for the developer.
